middle_end/flambda2/terms/static_const.ml

```diff
@@ -224,35 +220,29 @@ let apply_renaming t renaming =
       if or_var == or_var' then t else Boxed_nativeint or_var'
     | Mutable_string { initial_value = _ } | Immutable_string _ -> t
     | Immutable_float_block fields ->
-      let changed = ref false in
-      let fields =
-        List.map
-          (fun (field : _ Or_variable.t) ->
-            let field' : _ Or_variable.t =
-              match field with
-              | Var v -> Var (Renaming.apply_variable renaming v)
-              | Const _ -> field
-            in
-            if not (field == field') then changed := true;
-            field')
+      let fields' =
+        Misc.Stdlib.List.map_sharing
+          (fun (field : _ Or_variable.t) : _ Or_variable.t ->
+            match field with
+            | Var v ->
+              let v' = Renaming.apply_variable renaming v in
+              if v == v' then field else Var v'
+            | Const _ -> field)
```

Comment:
Maybe we should have a `Or_variable.map_sharing`/`Or_variable.apply_renaming`, so that this code is simpler.

Reply:
Done